Algorithms/Numeric/Decomposition/SVD.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using M = Utilities.Extensions.MatrixExtensions;
+using V = Utilities.Extensions.VectorExtensions;
+
+namespace Algorithms.Numeric.Decomposition
+{
+    /// <summary>
+    /// Singular Vector Decomposition decomposes any general matrix into its
+    /// singular values and a set of orthonormal bases.
+    /// </summary>
+    public static class Svd
```

Comment:
Maybe we should call it ThinSvd? Based on classification [here](https://en.wikipedia.org/wiki/Singular_value_decomposition#Reduced_SVDs)

Reply:
Yeah sounds good